test/TestAvailability.ts

```diff
@@ -0,0 +1,229 @@
+      const result = await happyTokenPoolDeployed.connect(ito_user).check_availability(invalidId);
+      await expect(result[0]).to.be.an("array").that.is.empty;
+    });
+
+    it("Should return status `started === true` when current time greater than start_time", async () => {
+      const fakeTime = (new Date().getTime() - 1000 * 10) / 1000;
+      creationParams.start_time = Math.ceil(fakeTime) - base_timestamp;
+      const { id: pool_id } = await getResultFromPoolFill(happyTokenPoolDeployed, creationParams);
+      const result = await getAvailability(happyTokenPoolDeployed, pool_id, user_address);
+
+      expect(result.started).to.be.true;
+    });
+
+    it("Should return status `started === false` when current time less than start_time", async () => {
+      const fakeTime = (new Date().getTime() + 1000 * 10) / 1000;
+      creationParams.start_time = Math.ceil(fakeTime) - base_timestamp;
+
+      const { id: pool_id } = await getResultFromPoolFill(happyTokenPoolDeployed, creationParams);
+      const result = await getAvailability(happyTokenPoolDeployed, pool_id, user_address);
+
+      expect(result.started).to.be.false;
+    });
+
+    it("Should return status `expired === true` when current time less than end_time", async () => {
+      const fakeTime = (new Date().getTime() - 1000 * 10) / 1000;
+      creationParams.end_time = Math.ceil(fakeTime) - base_timestamp;
+
+      const { id: pool_id } = await getResultFromPoolFill(happyTokenPoolDeployed, creationParams);
+      const result = await getAvailability(happyTokenPoolDeployed, pool_id, user_address);
+
+      expect(result.expired).to.be.true;
+    });
+
+    it("Should return status `expired === false` when current time less than end_time", async () => {
+      const fakeTime = (new Date().getTime() + 1000 * 10) / 1000;
+      creationParams.end_time = Math.ceil(fakeTime) - base_timestamp;
+
+      const { id: pool_id } = await getResultFromPoolFill(happyTokenPoolDeployed, creationParams);
+      const result = await getAvailability(happyTokenPoolDeployed, pool_id, user_address);
+
+      expect(result.expired).to.be.false;
+    });
+
+    it("Should return the same exchange_addrs which fill the happyTokenPoolDeployed", async () => {
+      const { id: pool_id } = await getResultFromPoolFill(happyTokenPoolDeployed, creationParams);
+      const result = await getAvailability(happyTokenPoolDeployed, pool_id, user_address);
+      expect(result.exchange_addrs).to.eql([eth_address, testTokenBDeployed.address, testTokenCDeployed.address]);
+    });
+
+    it("Should return the exchanged_tokens filled with zero when there was no exchange", async () => {
+      const { id: pool_id } = await getResultFromPoolFill(happyTokenPoolDeployed, creationParams);
+      const result = await getAvailability(happyTokenPoolDeployed, pool_id, user_address);
+      expect(result.exchanged_tokens.map((bn) => ethers.utils.parseEther(bn.toString()).toString())).to.eql([
```

Comment:
Looks wired. Why not use`result.exchanged_tokens.map((bn) => ethers.utils.formatEther(bn))`. see detail at <https://docs.ethers.io/v5/api/utils/display-logic/#utils-formatEther>